src/main.rs

```diff
@@ -230,3 +200,47 @@ pub fn init_logger() {
             .expect("Could not start logger");
     }
 }
+
+#[cfg(target_os = "windows")]
+fn windows_fix_dpi() {
+    use winapi::shared::windef::DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2;
+    use winapi::um::winuser::SetProcessDpiAwarenessContext;
+    unsafe {
+        SetProcessDpiAwarenessContext(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2);
+    }
+}
+
+#[cfg(target_os = "macos")]
+fn handle_macos() {
+    // incase of app bundle, we can just pass --disowned option straight away to bypass this check
+    #[cfg(not(debug_assertions))]
+    if !SETTINGS.get::<CmdLineSettings>().disowned {
+        if let Ok(curr_exe) = std::env::current_exe() {
+            assert!(std::process::Command::new(curr_exe)
+                .args(std::env::args().skip(1))
+                .arg("--disowned")
+                .spawn()
+                .is_ok());
+            std::process::exit(0);
```

Comment:
Now it calls exit() to stop process.

But i stil does not understand how it works?

Looks like, if process is called without `--disowned`, then it is called again with `--disowned`?

So, it's actually more like internal `flag`, not `option` to be used by user? And it always disowns process on macOS.

Reply:
Actually, this code should be platform-independent.

If found this issue https://github.com/neovide/neovide/issues/402, and for linux i have to do the same (use setsid --fork to fork process, so it would detach itself from terminal).

@Kethku In windows you say it is done by default?

https://github.com/neovide/neovide/issues/485

I found another issue, where it does not.

I think, right way to do this - is disown by default on all platforms, and rename flag to `--nofork` (just like gvim). This also will be less confusing to users (they will not try to use `--disowned` to disown process).

But that's for another PR i guess.

Reply:
I agree with this, and to be honest I don't fully understand this code either as it was contributed and I just scanned it for errors and tested.

I agree that for most cases it makes sense for disowned to be the default. My only concern is if somebody configures neovide to be the default git editor which waits for the application to finish, but like you say, this can be handled by just configuring the git editor to use --nofork

Reply:
Nice, i'll make another PR to refactor this.